Add Wagtail module for Social Media molecule

Scotchester opened the pull request:
This PR enables the Social Media molecule to be used as a module in Wagtail, available to the Sidefoot.


## Additions

- Ability to add Social Media module to Sidefoot streams in Wagtail


## Changes

- Tweak styling of the Social Media molecule


## Testing

1.  Pull down this branch
1. `gulp styles`
1. `./cfgov/manage.py migrate`
1. `./cfgov/manage.py runserver`
1. Login to your local Wagtail at http://localhost:8000/admin/
1. Add the new Social Media module to a page's Sidefoot
1. Preview that page
1. Also review an existing Blog page and the Footer to ensure the Social Media molecule there looks as expected.


## Review

- @anselmbradford 
- @KimberlyMunoz 
- @sebworks 
- @kave 


## Screenshots

![screen shot 2016-05-20 at 10 06 10](https://cloud.githubusercontent.com/assets/1044670/15430264/8dd0ebec-1e72-11e6-98ff-29254a2ea093.png)


## Notes

- I added 30px of top margin to `.post_body`, because bottom margin was removed from the Social Media molecule's atoms to be consistent to the atomic design spacing principles as I (and @anselmbradford) understand them. 30px seems consistent with the latest designs for articles, though those haven't been updated in quite a while. Please comment if you have any concerns with this. /cc @duelj @ajbush 


## Checklist

* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

sebworks commented on cfgov/jinja2/v1/_includes/molecules/social-media.html:
```diff
@@ -8,40 +8,47 @@
 {# TODO: Remove global page reference when moving to Wagtail. #}
-{% macro render( options={} ) %}
-{% set title = options.title | urlencode or ( page or {} ).seo_title | urlencode  %}
+{% set blurb = value.blurb | urlencode or
```
can you use parens and like ` value.blurb or page.seo_title or value.title or  'Look what I found on the CFPB’s site!' ) | urlencode?

Scotchester replied:
Ooh, good idea! I'll give that a try.

Scotchester replied:
That works, thanks!

anselmbradford commented on cfgov/jinja2/v1/_includes/molecules/social-media.html:
```diff
@@ -8,40 +8,47 @@
 
    ========================================================================== #}
 
+{# from 'macros/share.html' import share_twitter_url_new as _share_twitter_url #}
```
Should this line be here? Where's `share_twitter_url_new`? I don't see it in `share.html`. Also, `_share_twitter_url` is defined in this file at L:172. The underscore should imply it's private and not being imported from a public API of another file.

Scotchester replied:
Whoops! I had done that when trying to get it to work locally, because it seemed like having it defined in this file was causing me a problem, but in the end it wasn't. I just forgot to remove this commented line. Removed, along with the one a few lines below it.

anselmbradford commented on cfgov/v1/migrations/0086_auto_20160520_1351.py:
```diff
@@ -0,0 +1,29 @@
+        migrations.AlterField(
+            model_name='eventpage',
+            name='live_stream_url',
+            field=models.URLField(blank=True, help_text=b'Format: https://www.ustream.tv/embed/video_id.  It can be obtained by following the instructions listed here: https://support.ustream.tv/hc/en-us/articles/207851917-How-to-embed-a-stream-or-video-on-your-site', verbose_name=b'URL', validators=[django.core.validators.RegexValidator(regex=b'^https?:\\/\\/www\\.ustream\\.tv\\/embed\\/.*$')]),
```
Trivial: there's an extra space before `It can be obtained…` it looks like.

anselmbradford replied:
Hah, nvm, realized this is the migrations

anselmbradford commented:
When I deselect `Create sharing links` in wagtail it says it's required:

![screen shot 2016-05-20 at 5 23 11 pm](https://cloud.githubusercontent.com/assets/704760/15442492/8c9073c8-1eaf-11e6-80f2-c7adc633a92f.png)

Scotchester commented:
@anselmbradford Wow, hah, that was dumb of me. Should be fixed now!

anselmbradford commented:
👍  Works. A few enhancements if you see an easy way:
- Disable the input fields when `Create sharing links` is unchecked.
- Link to a list of language codes below the language input, such as https://dev.twitter.com/web/overview/languages

kave commented on cfgov/v1/migrations/0086_auto_20160520_1351.py:
```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
```
can we merge these 2 migrations into 1

Scotchester replied:
How do I do that?

kurtw replied:
Delete the migrations, then run `makemigrations` again. It will generate the one file that encompasses all the changes. Also, if you've already run these migrations locally, you can run `cfgov/manage.py migrate v1 --fake` after you make the one migration file to register that migration as already done.

kave commented:
👍 works fine, but i had a small comment on the email share button. Should that redirect ppl to the external site page? it technically would just bring up whatever email client they use locally... so its not really redirecting them anywhere?

Scotchester commented:
@kave Great point about the email link! I'll fix have a fix for that in soon.

Scotchester commented:
Updated with a fix for the email link issue. Had to force push, because I had rebased from the flapjack branch. Sorry if you had checked out this branch for testing!

Scotchester commented:
> Disable the input fields when Create sharing links is unchecked.

@kave or @kurtw Is this possible in Wagtail? Is there an existing example I can follow?

kurtw commented:
It's possible but out of scope for the PR. I'll respond with more details.

Scotchester commented:
Update pushed to squash migrations. I think this is good to go. One last thumbs-up for the new migrations?

kurtw commented:
All good here 👍

Scotchester commented:
Hmm... Travis is [reporting conflicting migrations](https://travis-ci.org/cfpb/cfgov-refresh/builds/132891990#L2818) but this branch is up-to-date with `flapjack` and auto-mergeable, so I'm not sure if I can do anything about that?

But

Scotchester commented:
I think I've finally fixed this. Let's see what Travis says.

Scotchester commented:
Hey! Travis likes it! At long last, here we go.
